docs/adoption.md

```diff
@@ -17,39 +19,243 @@ Clone/fork the following repositories:
+   `AZURE_SUBSCRIPTION_ID = Navigate to subscriptions and select the Subscription ID for your subscription `
+
+2. You will then reference this as environment variables in your github actions workflow. There will be an example provided further down which you can replicate. This allows the setup-terraform action to use the service principal credentials to provision your resources.
+
+3. Add STORAGE_ACCOUNT_KEY as a repository secret. This is the storage account key for your terraform backend. You can obtain this value if you navigate to Storage accounts, select the storage account for your terraform backend, select Access Keys, click on show keys and copy the top key value. (Note: These are rotating keys and are subject to change, this tutorial does not investigate how to work around this)
+
+4. Below is an example of how your workflow should look:
+
+```
+name: "Terraform"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  terraform:
+    name: "Terraform"
+    env:
+      ARM_CLIENT_ID: ${{ secrets.AZURE_AD_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.AZURE_AD_CLIENT_SECRET }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      ARM_TENANT_ID: ${{ secrets.AZURE_AD_TENANT_ID }}
+    runs-on: ubuntu-latest
+    environment: production
+
+    defaults:
+      run:
+        shell: bash
+        working-directory: './terraform'
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_wrapper: false
+
+      - name: Terraform fmt
+        run: terraform fmt -check
+        continue-on-error: true
+
+      - name: Terraform Init
+        run: terraform init -backend-config="storage_account_name=your_storage_account_name" -backend-config="resource_group_name=your_resource_group_name" -backend-config="container_name=tfstate" -backend-config="key=${{secrets.STORAGE_ACCOUNT_KEY}}"
+
+      - name: Terraform Validate
+        run: terraform validate -no-color
+
+      - name: Terraform Plan
+        run: terraform plan -var="storage_account_name=your_storage_account_name" -var="resource_group_name=your_resource_group_name" -var="container_name=tfstate" -var="key=${{secrets.STORAGE_ACCOUNT_KEY}}" -var="resource_group_location=your_resource_group_location"
+        continue-on-error: true
+
+      - name: Terraform Apply
+        run: terraform apply -auto-approve -var="storage_account_name=your_storage_account_name" -var="resource_group_name=your_resource_group_name" -var="container_name=tfstate" -var="key=${{secrets.STORAGE_ACCOUNT_KEY}}" -var="resource_group_location=your_resource_group_location"
+
+
+```
+
+_Note: the terraform commands with their respective flags should be in a single line_
+
+### Deploy housing-repairs-online-frontend
+
+Once you have completed the above steps, add the following terraform code to main.tf to provision a static web app as this is where the frontend is deployed.
```

Comment:
Given the following content is already present in the respective files, I don't feel it's necessary to include these instructions.